### ISSUE TITLE
Port labeling tasks to new architecture

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal#152]
Placeholder

**Comment by tcompa** [comment]
As of 0267a434feb3909a82e068a62ce8c991f6877c8c, `image_labeling.py` is essentially ready.
I tested it on a 2D dataset, using the per-FOV segmentation at pyramid level 3 (per-FOV Cellpose runtime is less than 2 seconds).

What's left in this issue:
1. I need to run `image_labeling` at least once on a 3D dataset.
2. The `Sandbox_image_labeling/pipeline.py` example includes a small hack, to replace the old `well` list in the metadata with a new one (where plates have a `_mip` suffix). This should be taken care of in the `replicate_zarr_structure` task.
3. I need to repeat everything for the `image_labeling_whole_well` task.

**Comment by jluethi** [comment]
Awesome!

Given https://github.com/fractal-analytics-platform/fractal/issues/149, we can eventually combine the `image_labeling` and `image_labeling_whole_well` tasks (because the well would also be a type of ROI). But if the porting of the existing `image_labeling_whole_well` is simple, we don't need to wait for #149 & this refactor of the labeling to get this in the current branch :)

**Comment by tcompa** [comment]
> The `Sandbox_image_labeling/pipeline.py` example includes a small hack, to replace the old `well` list in the metadata with a new one (where plates have a `_mip` suffix). This should be taken care of in the `replicate_zarr_structure` task.

This is fixed with 0267a434feb3909a82e068a62ce8c991f6877c8c

**Comment by tcompa** [comment]
Re: 3D-labeling

I have to go a very low resolution to make the run times reasonable on a CPU. Pyramid level 4 corresponds to FOV size `(10, 135, 160)`, leading to 30-40 seconds per FOV (with 400% CPU usage). In this case labeling results are obviously bad, but at least we are sure that the output has the correct shape. Here is how it looks:
![Screenshot from 2022-08-31 15-01-20](https://user-images.githubusercontent.com/3862206/187684439-3a9455af-48aa-4e60-bb3b-898fbbec2d67.png)

As a comparison, here is the output of 2D labeling on MIP data, at level 3:
![Screenshot from 2022-08-31 14-29-44](https://user-images.githubusercontent.com/3862206/187684519-7223eb5d-6202-4702-ae86-68805c57f426.png)

**Comment by tcompa** [comment]
Updated to-do list:
1. Update the MANIFEST with this new task;
2. Run it once through the server;
3. Also port `image_labeling_whole_well` (in principle that should be trivial, otherwise we may defer it to after #149).

**Comment by tcompa** [comment]
As of https://github.com/fractal-analytics-platform/fractal/commit/2d96523dbb4f2f2c8223c7cf24258f90e78bd7a5, we have the image-labeling task within the MANIFEST, and it is also part of the example workflow in `examples/Example_new_architecture`.

Notice that this example finds zero labels per-FOV, since the images are actually fake artificial images. But the correct behavior on real data was already verified above in this issue, and this test was mostly to check the fractal integration.

Still, I opened https://github.com/fractal-analytics-platform/fractal/issues/163 to have more realistic data available.

**Comment by tcompa** [comment]
As of 3c89a64b618a8ef2034efe33db46f80727b3e942, we have:
1. An updated version of `image_labeling_whole_well.py`, to work within the new architecture.
2. The usual temporary example file `tasks/Sandbox_image_labeling_whole_well/pipeline_2d.py`, that produces the output below (caveat: it requires an image folder that is not on the repo - see #163).
3. An up-to-date MANIFEST in `__init__.py`;
4. A new example in `examples/Example_new_architecture/user_dir/` (once again, this runs on fake images and finds 0 labels, but everything runs through until the end).

***

![Screenshot from 2022-08-31 16-47-59](https://user-images.githubusercontent.com/3862206/187710841-bc61a0fa-70fe-47dc-b453-94a5a8bc9b84.png)
